### PR TITLE
Apache HBase Alert Policies

### PR DESCRIPTION
**Pull request by schmikei**
# Alert Policies for Apache HBase in the Ops Agent

## Slow Operations

Slow operations can be an indication that the HDFS may be in an unhealthy state.

## HBase Server Stopped

If the HBase has unexpected downtime, this alert will help teams discern that the server is stopped and unable to function in their environment.

Required Log Query:

```sh
logName="projects/<project_id>/logs/hbase_system"
jsonPayload.message="regionserver.HeapMemoryManager: Stopping"
```

Replace <project_id> with your Project ID

## Authentication Errors

Authentication errors could indicate that clients do not have correct credentials or somebody is maliciously trying to access the HBase system.

### CODE REVIEW

File: alerts/hbase/slow-operations.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+    "displayName": "HBase - Slow Operations",
+    "documentation": {
+        "content": "Slow operations can be an indication that the HDFS may be in an unhealthy state. ",
```
Comment (adeychman) [inline alerts/hbase/slow-operations.json:4]: Can we share what is the threshold that needs to be breached for the slow operation alert to trigger?

File: alerts/hbase/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+Slow operations can be an indication that the HDFS may be in an unhealthy state.
+
+## HBase Server Stopped
+
+If the HBase has unexpected downtime, this alert will help teams discern that the server is stopped and unable to function in their environment.
+
+Required Log Query:
+
+```sh
```
Comment (adeychman) [inline alerts/hbase/README.md:13]: Is the `sh` cruft?

File: alerts/hbase/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Alert Policies for Apache HBase in the Ops Agent
+
+## Slow Operations
+
+Slow operations can be an indication that the HDFS may be in an unhealthy state.
+
+## HBase Server Stopped
+
+If the HBase has unexpected downtime, this alert will help teams discern that the server is stopped and unable to function in their environment.
```
Comment (adeychman) [inline alerts/hbase/README.md:9]: Can we share how long does it need to be down before the alert policy is triggered?

File: alerts/hbase/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+```sh
+logName="projects/<project_id>/logs/hbase_system"
+jsonPayload.message="regionserver.HeapMemoryManager: Stopping"
+```
+
+Replace <project_id> with your Project ID
+
+## Authentication Errors
+
+Authentication errors could indicate that clients do not have correct credentials or somebody is maliciously trying to access the HBase system.
```
Comment (adeychman) [inline alerts/hbase/README.md:22]: Will this trigger if 1 authentication error is present (i.e. through a Log Based Alert) or after a certain number of errors are present (i.e. through a metric threshold alert?) Could we please add the trigger information to the description? Thanks!

Comment (adeychman) [reply alerts/hbase/README.md:22]: And could the updates that are made in the ReadMe also be passed down to the alert policy JSON below. Thank you

File: alerts/hbase/authentication-errors.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    "displayName": "HBase - Authentication Errors",
+    "documentation": {
+        "content": "Authentication errors could indicate that clients do not have correct credentials or somebody is maliciously trying to access the HBase system. The alert is configured to fire if 5 authentication errors are happening over a minute. Feel free to modify to fit use cases.",
+        "mimeType": "text/markdown"
+    },
+    "userLabels": {},
+    "conditions": [
+        {
+            "displayName": "HBase Authentication Errors",
+            "evaluationMissingData": "EVALUATION_MISSING_DATA_NO_OP",
```
Comment (chanseokoh) [inline alerts/hbase/authentication-errors.json:11]: Property "evaluationMissingData" does not exist in object specification.

**Review by chanseokoh — APPROVED** [review]

**Pull request merged**